### PR TITLE
fix(Dashboard): Copying a Dashboard does not commit the transaction

### PR DESCRIPTION
**Pull request by geido**
### SUMMARY
We noticed that when copying a Dashboard the transaction wasn't being committed. This PR introduces a command for the dashboard copy operation and uses the transaction decorator to commit the transaction.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
N.A.

### TESTING INSTRUCTIONS
1. Copy a Dashboard
2. The Dashboard should be copied successfully

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by geido** [comment]
> @geido Would you mind also fixing `delete_embedded` and `copy_dash`?

I will in follow-ups.

**Review by kgabryje — APPROVED** [review]
Looks good!

File: superset/commands/dashboard/copy.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+logger = logging.getLogger(__name__)
+
+
+class CopyDashboardCommand(BaseCommand):
+    def __init__(self, original_dash: Dashboard, data: dict[str, Any]) -> None:
+        self._original_dash = original_dash
+        self._properties = data.copy()
+
+    @transaction(on_error=partial(on_error, reraise=DashboardInvalidError))
```
Comment (michael-s-molina) [inline superset/commands/dashboard/copy.py:39]: Could you reraise using a more generic exception such as `DashbordCopyError` (will need to create)? reraise means catching all exceptions and re-raising them as a generic one. For example, you might have `DashboardForbiddenError`.

File: superset/dashboards/api.py [hunk 2]
```diff
@@ -1606,9 +1607,11 @@ def copy_dash(self, original_dash: Dashboard) -> Response:
             return self.response_400(message=error.messages)
 
         try:
-            dash = DashboardDAO.copy_dashboard(original_dash, data)
+            dash = CopyDashboardCommand(original_dash, data).run()
         except DashboardForbiddenError:
             return self.response_403()
+        except DashboardInvalidError:
+            return self.response_400()
```
Comment (michael-s-molina) [inline superset/dashboards/api.py:1614]:
```suggestion
        except DashboardCopyError:
            return self.response_403()
```

**Comment by sadpandajoe** [comment]
@supersetbot label 4.1

**Review by michael-s-molina — APPROVED** [review]

**Comment by sadpandajoe** [comment]
@supersetbot label 4.1

**Pull request merged**